### PR TITLE
JavaScriptFragment.asJson()

### PR DESCRIPTION
**Pull request by labkey-matthewb**
replace Jsp.unsafe() with JavaScriptFragment.unsafe() in <script> blocks replace new ObjectMapper() with JsonUtil.DEFAULT_MAPPER or JsonUtil.createDefaultMapper()

#### Related Pull Requests
* 

#### Changes
* eliminate use of new ObjectMapper()
* new JavaScriptFragment helpers: asJson() asString()
* remove usages of unsafe()/text() from <script> blocks (replace with q() or JavaScriptFragment)

### CODE REVIEW

File: mothership/src/org/labkey/mothership/MothershipManager.java [hunk 1]
```diff
@@ -388,7 +389,7 @@ private String getBestJson(String currentValue, String newValue, String serverSe
         }
 
         // Rather than overwrite the current json map, merge the new with the current.
-        ObjectMapper mapper = new ObjectMapper();
+        ObjectMapper mapper = JsonUtil.createDefaultMapper();
```
Comment (labkey-adam) [inline mothership/src/org/labkey/mothership/MothershipManager.java:392]: Could be `JsonUtil.DEFAULT_MAPPER`

File: study/src/org/labkey/study/view/datasetVisibility.jsp [hunk 2]
```diff
@@ -244,7 +245,7 @@
 
     Ext4.onReady(function()
     {
-        var datasetInfo = <%=unsafe(JsonUtil.DEFAULT_MAPPER.writeValueAsString(datasetInfo))%>;
+        var datasetInfo = <%=JavaScriptFragment.asJson(datasetInfo)%>;
         var store = LABKEY.study.DataViewUtil.getViewCategoriesStore({
             storeId : '<%=h(storeId)%>',
```
Comment (labkey-adam) [inline study/src/org/labkey/study/view/datasetVisibility.jsp:250]: That looks wrong

**Review by labkey-adam — CHANGES_REQUESTED** [review]
There's an h() to q() conversion in there somewhere

File: study/src/org/labkey/study/view/datasetVisibility.jsp (hunk 2 above)
Comment (labkey-adam) [reply study/src/org/labkey/study/view/datasetVisibility.jsp:250]: Meaning line 250

File: mothership/src/org/labkey/mothership/MothershipManager.java (hunk 1 above)
Comment (labkey-matthewb) [reply mothership/src/org/labkey/mothership/MothershipManager.java:392]: I've taken to using DEFAULT_MAPPER when I can inline the whole usage.  Just a style choice and speeds up converting code (what does readerForUpdating() do?  beats me.).

File: study/src/org/labkey/study/view/datasetVisibility.jsp (hunk 2 above)
Comment (labkey-matthewb) [reply study/src/org/labkey/study/view/datasetVisibility.jsp:250]: I assume this is some Ext weirdness where you have to htmlEncode() id's so they don't get injected into the html?  I don't think I changed this.

Comment (labkey-matthewb) [reply study/src/org/labkey/study/view/datasetVisibility.jsp:250]: configRequestabilityRules.jsp has a strange use of h() which I changed to qh(), not sure if it's wrong.

**Review by labkey-adam — APPROVED** [review]

**Pull request merged**